### PR TITLE
Run basic gost engine tests

### LINKED ISSUE #14265: Re-instantiate gost engine tests in CI
GOST engine tests provide a good coverage for ENGINE mechanism and should be added to the CI scenarios, at least in minimal form.

### PR DESCRIPTION
**Pull request by beldmit**
CI scenario to run minimal gost-engine tests. Fixes #14265 

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -146,6 +146,28 @@ jobs:
...
+  buildtest-gost:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: config
+      run: ./config --strict-warnings && perl configdata.pm --dump
+    - name: make
+      run: make -s -j4
+    - name: checkout gost engine
+      run: git clone --depth 1 -b master https://github.com/gost-engine/engine.git gost-engine-test
```
Comment (levitte) [inline .github/workflows/ci.yml:160]: "actions/checkout@v2" is quite useful for this!

``` yaml
    - name: checkout gost engine
      uses: actions/checkout@v2
      with:
        repository: gost-engine/engine
        ref: master
        path: gost-engine-test
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -146,6 +146,28 @@ jobs:
     - name: make test
       run: make test HARNESS_JOBS=${HARNESS_JOBS:-4}
 
+        #TODO 1. Install Test::V0 to run perl tests
```
Comment (levitte) [inline .github/workflows/ci.yml:149]: Here's the recommended step:

``` yaml
    - name: install cpanm and Test2::V0
      uses: perl-actions/install-with-cpanm@v1
      with:
        install: Test2::V0
```

**Review by levitte — CHANGES_REQUESTED** [review]

**Comment by t8m** [comment]
Dmitry, I am working on the same thing right now already.  We want to eventually run all the external tests (of course after making them to pass).

File: .github/workflows/ci.yml (hunk 1 above)
Comment (beldmit) [reply .github/workflows/ci.yml]: Done. Thanks!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (beldmit) [reply .github/workflows/ci.yml]: Done. Many thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (t8m) [reply .github/workflows/ci.yml]: Also IMO we should use known-good commit id otherwise we will regress if something is broken in gost-engine master.

**Comment by beldmit** [comment]
@t8m, do you plan to add those tests run explicitly, as they were before?

**Comment by t8m** [comment]
Yes, I wanted to use the existing scripts for external tests.

**Comment by beldmit** [comment]
I'd prefer to add gost tests on regular basis. Gost-engine allows to catch _strange_ breakage when it happens. 
I also don't like the submodule stuff

**Comment by t8m** [comment]
Well we have the submodule stuff so if we want to give up on using it we should delete the submodules. But anyway we should use some known good commit ids when running the tests. We cannot blindly follow the master branches of those external projects.

**Comment by beldmit** [comment]
I agree about the known commit and will specify it, but first let it run with master correctly

**Comment by beldmit** [comment]
The specific commit id is added. I'm not sure what to do on API change, but in other cases we will immediately catch a regression.

**Comment by levitte** [comment]
A benefit with the submodule approach is that it forces using known-good commit ids without having to update the CI scripts...  I do think it should be considered.  It also means that anyone can test at home without having to go look at the workflow files to figure out how to do it.

**Comment by beldmit** [comment]
I agree. But explicit running the external tests leads to the situation when they aren't run at all. 

I'd prefer to fix the GOST engine on API changes, but to be sure that the regression is not introduced.

**Comment by t8m** [comment]
Not sure what do you exactly mean by running the tests explicitly. The plan is to run them on every CI run unless they are too slow.

**Comment by levitte** [comment]
> Not sure what do you exactly mean by running the tests explicitly. The plan is to run them on every CI run unless they are too slow.

You can do it now:

``` console
$ cd $OPENSSL3SRC
$ git submodule update --init gost-engine
$ cd $OPENSSL3BUILD
$ $OPENSSL3SRC/Configure --debug
$ make test TESTS=test_external_gost_engine
```

**Comment by levitte** [comment]
(the external gost test needs a bit of an update)

**Comment by t8m** [comment]
This is alternative PR: #14416

**Comment by beldmit** [comment]
Closing. Many thanks @t8m!

**Pull request closed without merging**